### PR TITLE
Auto-generate fog events if there are none

### PR DESCRIPTION
**Pull request by raphaelgoulart**
image
image
This PR autogenerates the aforementioned FogOn/FogOff events; it happens before the venue lighting autogeneration, since the venue might be authored but still not have any fog events.
Tested by @TheFatBastid

### CODE REVIEW

File: Assets/Script/Settings/VenueAutoGenerationPreset.cs [hunk 1]
```diff
@@ -406,5 +408,50 @@ private CameraPacingPreset StringToCameraPacing(string cameraPacing)
...
+            return false;
+        }
+
+        public SongChart GenerateFogEvents(SongChart chart)
+        {
+            var lastTick = chart.GetLastTick();
+            var resolution = chart.Resolution;
+            var startInterval = 8 * 4;
+            var fogOnInterval = 32 * 4;
+            var fogOffInterval = 8 * 4;
```
Comment (EliteAsian123) [inline Assets/Script/Settings/VenueAutoGenerationPreset.cs:428]: Can be converted into constants

File: Assets/Script/Settings/VenueAutoGenerationPreset.cs [hunk 2]
```diff
@@ -406,5 +408,50 @@ private CameraPacingPreset StringToCameraPacing(string cameraPacing)
...
+
+            uint nextTick = (uint)(resolution * startInterval);
```
Comment (EliteAsian123) [inline Assets/Script/Settings/VenueAutoGenerationPreset.cs:430]: Space after casts

**Review by EliteAsian123 — CHANGES_REQUESTED** [review]
Overall very nice, just a few minor styling changes

**Comment by raphaelgoulart** [comment]
Committed minor styling changes

**Pull request merged**